### DISCUSSION TITLE
Loading SpreadsheetView programmatically

### DISCUSSION THREAD

**Discussion by adamlabadorf** [GMOD/jbrowse-components#2424] [Q&A]
Hi,

How can I make a spreadsheet view with tabular data available when building a jbrowse instance, rather than loading it in through the browser? I'd like to build this in so collaborators can easily navigate some results without sharing it with them separately. Thanks!

**Comment by cmdcolin** [comment]
One way to do this is to create a share link to an instance of the spreadsheet view with your data loaded. There might be other ways also, but that is probably the easiest right now

**Reply by cmdcolin** [reply]
have also considered letting admins set up 'session template' in their instance so that it is easier to do this without making explicit share links https://github.com/GMOD/jbrowse-components/issues/1384

**Reply by adamlabadorf** [reply]
Thanks for this idea. In our case we use [Open OnDemand](https://openondemand.org/) with to interact with our cluster. This makes it trivial to provide my collaborators access to JBrowse instances without having a dedicated web server (which is amazingly useful!), but session sharing links don't work. We also build these instances as part of pipelines, so it would be nice to be able to 'push the button' and have everything update in the general case. Even if it's complicated, is there a way to do this under the hood?

**Reply by cmdcolin** [reply]
Hmm. this is a good question. One way to do it would be to write a plugin that programmatically starts up a SpreadsheetView on startup

I tried to hack up a quick method to do this here https://github.com/cmdcolin/jbrowse-plugin-autocreatespreadsheet

**Reply by cmdcolin** [reply]
It is a quick demo but it does automatically launch a VCF in a spreadsheet view on startup when the plugin is loaded

![localhost_3000__config=http%3A%2F%2Flocalhost%3A9000%2Fjbrowse_config json session=local-V2RTG6Q4E](https://user-images.githubusercontent.com/6511937/139345491-c146ec44-0cc6-4508-9363-7faf0a8fb9ef.png)

**Reply by cmdcolin** [reply]
let me know if you have questions about the plugin development or setup, it can be a little tricky but happy to help
